internal/config/application.go

```diff
@@ -72,10 +72,13 @@ func (cfg Application) ToCatalogerConfig() cataloger.Config {
 		},
 		Catalogers:  cfg.Catalogers,
 		Parallelism: cfg.Parallelism,
-		Golang: golangCataloger.GoCatalogerOpts{
-			SearchLocalModCacheLicenses: cfg.Golang.SearchLocalModCacheLicenses,
-			LocalModCacheDir:            cfg.Golang.LocalModCacheDir,
-		},
+		Golang: golangCataloger.NewGoCatalogerOpts(
```

Comment:
we've been starting to lean into a struct builder pattern in places instead of package-wide functional options in a few places. They have the advantage of not polluting the package namespace with options for multiple structs. I think we should lean in favor of that for now:
```
golangVataloger.NewGoCatalogerConfig().
  WithSearchLocalModCacheLicenses(...).
  WithProxy(...).
  ...
```